src/modules/collection/sagas.ts

```diff
@@ -420,6 +410,31 @@ function* handleTransactionSuccess(action: FetchTransactionSuccessAction) {
   }
 }
 
+/**
+ * Post proccesses a collection that was published to the blockchain by singaling the
+ * builder server that the collecton has been published, setting the item ids,
+ * deploys the item entities to the Catalyst sever and creates the forum post.
+ *
+ * @param collection - The collection to post process.
+ */
+function* postProcessCollectionTransaction(collection: Collection) {
+  const avatarName: string | null = yield select(getName)
```

Comment:
I really would prefer that we didn't do any selects after dispatching actions in a sagas as it can lead to really complex side effects that are hard to debug. I moved it upwards to keep it before any other generator / async function that could change it. TBH, I would move these selectors to the functions that use `postProcessCollectionTransaction`, but `deployItems` and `publishCollection` already use selectors.

Reply:
100%, my bad. Did not realize that we where doing dispatches above the if I referenced! this is 💯